core/app/models/spree/order/checkout.rb

```diff
@@ -221,7 +221,15 @@ def update_from_params(params, permitted_params, request_env = {})
             run_callbacks :updating_from_params do
               # Set existing card after setting permitted parameters because
               # rails would slice parameters containg ruby objects, apparently
-              existing_card_id = @updating_params[:order] ? @updating_params[:order].delete(:existing_card) : nil
+
+              payment_method = if @updating_params[:order] && @updating_params[:order][:payments_attributes]
```

Comment:
We can use something like this, if we are sure that `order` is always a hash.
```suggestion
              payment_method = if @updating_params.dig(:order, :payments_attributes)
```